op-chain-ops/genesis/layer_two.go

```diff
@@ -54,18 +58,51 @@ func BuildL2Genesis(config *DeployConfig, l1StartBlock *types.Block) (*core.Gene
 			log.Warn("Skipping disabled predeploy.", "name", name, "address", predeploy.Address)
 			continue
 		}
+
 		codeAddr := predeploy.Address
-		if !predeploy.ProxyDisabled {
-			codeAddr, err = AddressToCodeNamespace(predeploy.Address)
+		switch name {
+		case "Permit2":
+			deployerAddress, err := bindings.GetDeployerAddress(name)
+			if err != nil {
+				return nil, err
+			}
+			deployerAddressPtr := common.BytesToAddress(deployerAddress)
```

Comment:
This technically isn't a pointer. I don't think its a good practice to name variables based on the type